distributed-commandbus-springcloud/src/main/java/org/axonframework/springcloud/commandhandling/SpringCloudCommandRouter.java

```diff
@@ -418,4 +374,137 @@ private ConsistentHash suspect(Member member) {
+         */
+        public Builder routingStrategy(RoutingStrategy routingStrategy) {
+            assertThat(routingStrategy,
+                       Objects::nonNull,
+                       () -> new AxonConfigurationException("RoutingStrategy may not be null"));
+            this.routingStrategy = routingStrategy;
+            return this;
+        }
+
+        /**
+         * Sets a {@link Predicate} of generic type {@link ServiceInstance}, used to filter out ServiceInstances from
+         * the membership update loop. Defaults to the
+         * {@link SpringCloudCommandRouter#serviceInstanceMetadataContainsMessageRoutingInformation(ServiceInstance)},
+         * which filters any instances missing message routing information key-value pairs in their metadata.
+         *
+         * @param serviceInstanceFilter the {@link Predicate} of generic type {@link ServiceInstance}, used to filter
+         *                              out ServiceInstances from the membership update loop
+         * @return the current Builder instance, for a fluent interfacing
+         */
+        public Builder serviceInstanceFilter(Predicate<ServiceInstance> serviceInstanceFilter) {
+            assertThat(serviceInstanceFilter,
+                       Objects::nonNull,
+                       () -> new AxonConfigurationException("ServiceInstanceFilter may not be null"));
+            this.serviceInstanceFilter = serviceInstanceFilter;
+            return this;
+        }
+
+        /**
+         * Sets the {@link ConsistentHashChangeListener} which is notified when a change in membership has
+         * <em>potentially</em> caused a change in the consistent hash. Defaults to a no-op solution.
+         *
+         * @param consistentHashChangeListener the {@link ConsistentHashChangeListener} which is notified when a change
+         *                                     in membership has <em>potentially</em> caused a change in the consistent
+         *                                     hash
+         * @return the current Builder instance, for a fluent interfacing
+         */
+        public Builder consistentHashChangeListener(ConsistentHashChangeListener consistentHashChangeListener) {
+            assertThat(consistentHashChangeListener,
+                       Objects::nonNull,
+                       () -> new AxonConfigurationException("ConsistentHashChangeListener may not be null"));
+            this.consistentHashChangeListener = consistentHashChangeListener;
+            return this;
+        }
+
+        /**
+         * Initializes a {@link SpringCloudCommandRouter} as specified through this Builder.
+         *
+         * @return a {@link SpringCloudCommandRouter} as specified through this Builder
+         */
+        public SpringCloudCommandRouter build() {
+            return new SpringCloudCommandRouter(this);
+        }
+
+        private void validate() {
+            assertThat(discoveryClient,
```

Comment:
Extraction to a separate method.

Reply:
Replaced for the `BuilderUtils` functions.
What do you think?